src-electron/db/query-package.js

```diff
@@ -690,23 +690,52 @@ async function getAllPackages(db) {
     )
     .then((rows) => rows.map(dbMapping.map.package))
 }
+/**
+ * Retrieves attribute access interface options from the database.
+ *
+ * This function performs a complex query to fetch options related to a specific code and package ID. It combines results from
+ * the PACKAGE_OPTION table with those from ATTRIBUTE and CLUSTER tables using a UNION. The purpose is to gather a comprehensive
+ * list of options that include both direct package options and those inferred from attributes' storage policies and their associated
+ * clusters.
+ *
+ * @param {Object} db - The database connection object.
+ * @param {string} code - The option code or storage policy code to query for.
+ * @param {number} packageId - The ID of the package to which the options are related.
+ * @returns {Promise<Array>} A promise that resolves to an array of option objects, each containing the option category, code, and label.
+ */
+async function getAttributeAccessInterface(db, code, packageId) {
+  try {
+    const extendedQuery = `
+      SELECT
```

Comment:
> Add a comment stating that this query is used to get the global attributes of clusters which cannot be gathered by the subsequent query since that only gathers access interface of attributes with a cluster reference.
> 
> 
> 
> I am confused on why we would want this interface in the attribute table and also in the zcl.json(Option_category). Seems like this data should just be in the attribute table(xml). May be with exceptions to global attributes but ideally it should all be in the xml and not package_option.

It cannot be in the Attribute table because that would conflate the ideas. They are global attributes.

The rules are not in the spec. They are driven by the SDK so they should not be included in a table that represents spec data. 

Hope that clears it up